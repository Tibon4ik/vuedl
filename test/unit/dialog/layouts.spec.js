import Dialog from '../../../src/dialog'
// import Vue from 'vue'
// import DialogView from '../../../src/components/view'
// import { mount } from '../../utils'

describe('dialog layouts', () => {
  beforeAll(() => {
    // Vue.component('dialog-view', DialogView)
    // Dialog.prototype.context = context
  })

  // test('Check default layouts when not sended', async () => {
  //   const dlg = new Dialog({
  //     template: '<p>foo</p>'
  //   })
  //   // expect(dlg._wrapper).toHaveProperty('component')
  //   // await dlg.show()
  //   // expect(dlg.element).toMatchSnapshot()
  //   // // expect(asyncData).toHaveBeenCalledTimes(1)
  //   // // expect(dialog.vmd.$data).toHaveProperty('msg', 'foo')
  //   // expect(dlg.innerHTML).toBe('<p>foo</p>')
  //   // dialog.close()
  // })

  test('check error without component', () => {
    expect(() => {
      new Dialog() // eslint-disable-line no-new
    }).toThrow()
  })

  test('Check layout', async () => {
    const layout = {
      template: '<div><slot/></div>'
    }

    const dialog = new Dialog({
      template: '<p>foo</p>'
    }, {
      layout: {component: layout}
    })
    await dialog.show()
    expect(dialog.element).toMatchSnapshot()
    expect(dialog.element.textContent).toBe('foo')
    dialog.close()
  })

  test('Check layout object', async () => {
    const dialog = new Dialog({
      template: '<p>foo</p>',
      layout: {
        template: '<div><slot/></div>'
      }
    })
    await dialog.show()
    expect(dialog.element).toMatchSnapshot()
    expect(dialog.element.textContent).toBe('foo')
    dialog.close()
  })

  /* test('Check layout slots', async () => {
    const layout = {
      template:
        `<div>
          <slot name="header"/>
          <p>above content</p>
          <slot/>
          <p>below content</p>
          <slot name="footer"/>
        </div>`
    }

    const dialog = new Dialog({
      template:
      `<div>
          <template slot="header">header</template>
          <div slot="footer">footer content</div>
          content
        </div>`
    }, {
      layout
    })
    await dialog.show()

    expect(dialog.element).toMatchSnapshot()
    dialog.close()
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  }) */
})
