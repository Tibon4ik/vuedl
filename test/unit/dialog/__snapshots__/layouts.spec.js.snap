// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog layouts Should close when in not persistent and not loading state 1`] = `
<div
  class="dialog-layout"
>
  <div>
    <p>
      foo
    </p>
  </div>
</div>
`;

exports[`dialog layouts Should work with layout 1`] = `
<div
  class="dialog-layout"
>
  <div>
    <p>
      foo
    </p>
  </div>
</div>
`;

exports[`dialog layouts Should work with layout with component property 1`] = `
<div>
  <div>
    <p>
      foo
    </p>
  </div>
</div>
`;
